source/js/viewer-core.js

```diff
@@ -404,7 +404,8 @@ function ViewerCore(element, options, publicInstance)
             var options = {
                 viewport: viewerState.viewport,
                 outerElement: viewerState.outerElement,
-                innerElement: viewerState.innerElement
+                innerElement: viewerState.innerElement,
+                enableCrossOrigin: settings.options.enableCrossOrigin
```

Comment:
I changed the viewer core to use the viewer state and options to get the enableCrossOrigin root setting